langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiEmbeddingModel.java

```diff
@@ -0,0 +1,92 @@
+package dev.langchain4j.model.azure;
+
+import dev.ai4j.openai4j.OpenAiClient;
+import dev.ai4j.openai4j.embedding.EmbeddingRequest;
+import dev.ai4j.openai4j.embedding.EmbeddingResponse;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.Tokenizer;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.model.embedding.TokenCountEstimator;
+import lombok.Builder;
+
+import java.net.Proxy;
+import java.time.Duration;
+import java.util.List;
+
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static java.time.Duration.ofSeconds;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Represents a connection to the OpenAI embedding model, such as text-embedding-ada-002, accessing it via Azure.
```

Comment:
Represents a connection to the OpenAI embedding model, hosted on Azure (like text-embedding-ada-002).